compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -355,25 +365,36 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         {
             self.suggest_missing_writer(rcvr_ty, args)
         } else {
-            struct_span_err!(
-                tcx.sess,
-                span,
-                E0599,
-                "no {} named `{}` found for {} `{}` in the current scope",
-                item_kind,
-                item_name,
-                rcvr_ty.prefix_string(self.tcx),
-                ty_str_reported,
-            )
+            let mut err = tcx.sess.struct_span_err_with_code(span, fluent::hir_typeck_no_associated_item, rustc_errors::error_code!(E0599));
+            err.set_arg("item_kind", item_kind)
+            .set_arg("item_name", item_name)
+            .set_arg("ty_prefix", if trait_missing_method {
+                // FIXME(mu001999) E0599 maybe not suitable here because it is for types
+                Cow::from("trait")
+            } else {
+                rcvr_ty.prefix_string(self.tcx)
+            })
+            .set_arg("ty_str", ty_str_reported)
+            .set_arg("postfix", if trait_missing_method {
+                ""
+            } else {
+                " in the current scope"
+            });
```

Comment:
This is not the best way to do this, since the string would be in english. You should set a bool and match that in the fluent resource.